lsi/rom/tools/cp16mic.py

```diff
@@ -0,0 +1,981 @@
+        '*': (6, lambda x, y: x * y),   # binary mul
+        '/': (6, lambda x, y: x // y),  # binary div
+        '+': (5, lambda x, y: x + y),   # binary add
+        '-': (5, lambda x, y: x - y),   # binary sub
+        '&': (4, lambda x, y: x & y),   # binary and
+        '^': (3, lambda x, y: x ^ y),   # binary xor
+        '|': (2, lambda x, y: x | y)    # binary or
+    }
+    EXP_UNA = '+-~'
+    EXP_OPS = '+-~*/+-&^|'
+    EXP_ALL = '+-~*/+-&^|()'
+ 
+    def __init__(self):
+        self.ps = 0             # pass number
+        self.lc = 0             # location counte
+        self.radx = 8           # default radix
+        self.lnum = 0           # line number
+        self.fnam = None        # source file name
+        self.fend = 0           # .end directive found
+        self.wcnt = 0           # warning counter
+        self.ecnt = 0           # error counter
+        self.elst = []          # line error list for listing
+        self.efst = 0           # first error output
+        self.bloc = {}          # local labels block
+        self.cloc = {}          # current local block
+        self.clin = {}          # local block line number
+        self.gval = None        # generated value for listing
+        self.data = [-1] * MAX_LC
+        self.symb = {
+            'G'     : (0, T_REG),
+	        'GL'    : (0, T_REG),
+	        'GH'    : (1, T_REG),
+	        'RBA'   : (2, T_REG),
+	        'RBAL'  : (2, T_REG),
+	        'RBAH'  : (3, T_REG),
+	        'RSRC'  : (4, T_REG),
+	        'RSRCL' : (4, T_REG),
+	        'RSRCH' : (5, T_REG),
+	        'RDST'  : (6, T_REG),
+	        'RDSTL' : (6, T_REG),
+	        'RDSTH' : (7, T_REG),
+	        'RIR'   : (8, T_REG),
+	        'RIRL'  : (8, T_REG),
+	        'RIRH'  : (9, T_REG),
+	        'RPSW'  : (10, T_REG),
+	        'RPSWL' : (10, T_REG),
+	        'RPSWH' : (11, T_REG),
+	        'SP'    : (12, T_REG),
+	        'SPL'   : (12, T_REG),
+	        'SPH'   : (13, T_REG),
+	        'PC'    : (14, T_REG),
+	        'PCL'   : (14, T_REG),
+	        'PCH'   : (15, T_REG),
+	        'I4'    : (1, T_SYM),
+	        'I5'    : (2, T_SYM),
+	        'I6'    : (4, T_SYM),
+    	    'C'     : (1, T_SYM),
+	        'V'     : (2, T_SYM),
+	        'Z'     : (4, T_SYM),
+    	    'N'     : (8, T_SYM),
+	        'C8'    : (16, T_SYM),
+	        'C4'    : (32, T_SYM),
+	        'ZB'    : (64, T_SYM),
+	        'NB'    : (128, T_SYM),
+	        'UB'    : (0, T_SYM),
+	        'LB'    : (1, T_SYM),
+	        'UBC'   : (2, T_SYM),
+	        'LBC'   : (3, T_SYM),
+	        'RMW'   : (4, T_SYM),
+	        'TG6'   : (1, T_SYM),
+	        'TG8'   : (2, T_SYM),
+	        'LRR'   : (1, T_SYM),
+	        'RSVC'  : (2, T_SYM)
+        }
+        return
+
+    def dir_title(self, opcode, word):
+        #
+        # Ignore .title for compatibility
+        #
+        return -1
+
+    def dir_radix(self, opcode, word):
+        radx = self.eval_exp(opcode[1])
+        if radx not in (8, 10):
```

Comment:
why not radix16 there?

Reply:
radix-16 is ambigous
hard to distinguish valid names and digits: BAC0 - might be name or digit

Reply:
I see